src/schema/v2/lot.ts

```diff
@@ -0,0 +1,287 @@
+import { GraphQLObjectType, GraphQLString } from "graphql"
+import { reduce } from "lodash"
+import { pageable } from "relay-cursor-paging"
+import { ResolverContext } from "types/graphql"
+import { connectionWithCursorInfo } from "./fields/pagination"
+import { SaleArtworkType } from "./sale_artwork"
+
+const LotType = new GraphQLObjectType<any, ResolverContext>({
+  name: "Lot",
+  fields: () => ({
+    internalID: {
+      type: GraphQLString,
+      resolve: ({ saleArtwork }) => saleArtwork._id,
+    },
+    saleArtwork: {
+      type: SaleArtworkType,
+      resolve: ({ saleArtwork }) => {
+        return saleArtwork
+      },
+    },
+    // internalID: ({saleArtwork}) => saleArtwork.id
+    // ...InternalIDFields,
+  }),
+})
+
+export const auctionLotConnection = connectionWithCursorInfo({
+  nodeType: LotType,
+})
+
+export const watchedLotsConnection = {
+  description: "A list of lots a user is watching",
+  type: auctionLotConnection.connectionType,
+  args: pageable(),
+  resolve: async (_parent, _args, context) => {
+    // rough approach:
+
+    // get Sale Artworks
+    // query gravity api/v2/sale_artworks?included_watched_artworks=true
+
+    // get lot state for each sale artwork from causality (ex query:
+    // query {
+    //   lots( ids: saleArtworks.map(sa => sa.id) ) {
+    //    # We probably have to select all graphql fields here?
+    //   }
+    // }
+
+    // zip them together and return with full necessary relay fields
+    const lotData = [lotStub]
+    const lotDataMap = reduce(
+      lotData,
+      (acc, lot) => {
+        acc[lot.internalID] = lot
+        return acc
+      },
+      {}
+    )
+
+    context.lotDataMap = lotDataMap
```

Comment:
For the record, I'm fine with using `context` to persist this for the request. It can be a nice and easy workaround from other awkward approaches to thread through or propagate some data throughout the request stack, possibly in a more proper GraphQL-y/stitching way.

An alternate approach to avoid this, might be:
  - declare `watchedLotConnection` in the causality stitching file
  - you'll be able to do the correct `lot: AuctionStateLotState` (or whatever it is) there
  - the parent resolver defined for the connection can essentially do what you have here, fetch and zip both sources of data. possibly no need to use `context`.
  - the node resolvers for the connection might then be able to access the `lot` from their parents

Not sure if you tried that out, but it might be worth a shot.

cc @zephraph

Reply:
Yeah, that seems possible! I'm having trouble visualizing what that'd look like though.

Reply:
Here's an example of what I mean: https://github.com/artsy/metaphysics/compare/master...mzikherman:xtina-starr-query-watched-lots

Basically moving all the type defs and resolvers 'up' a level into the stitching file, vs. mixing and matching the MP and Causality types.

Reply:
I gotcha. Yep, that makes sense.

Reply:
oh, i had just started a quick spike on this - i'll take a look at the branch.

Reply:
alright, 9e4aab4ba0d22d4301961270c9feecb32761c16d accomplishes this and also adds the sale artwork loader so actual watched lots will appear from the user.